plugins/module_utils/common/api/MIGRATION_EXAMPLE.md

```diff
@@ -0,0 +1,411 @@
+
+    @property
+    def path(self) -> str:
+        # Validate required parameters when path is accessed
+        if self.fabric_name is None:
+            raise ValueError("fabric_name must be set before accessing path")
+
+        segments = [self.fabric_name, "config-deploy"]
+        if self.switch_id:
+            segments.append(self.switch_id)
+
+        base_path = BasePath.control_fabrics(*segments)
+
+        # Build composite query string
+        composite = CompositeQueryParams()
+        composite.add(self.query_params)
+
+        query_string = composite.to_query_string()
+        if query_string:
+            return f"{base_path}?{query_string}"
+        return base_path
+
+    @property
+    def verb(self) -> Literal["POST"]:
+        return "POST"
+```
+
+**Usage (Full Property-Style Interface):**
+
+```python
+# Create empty endpoint
+request = EpFabricConfigDeploy()
+
+# Set path parameters via properties
+request.fabric_name = "MyFabric"
+request.switch_id = ["CHM1234567", "CHM7654321"]
+
+# Set query parameters via properties
+request.query_params.force_show_run = True
+request.query_params.include_all_msd_switches = True
+
+path = request.path
+verb = request.verb
+```
+
+**Benefits:**
+
+- **Consistent Interface**: Both path and query parameters use property-style interface
+- **Separation of Concerns**: Path parameters vs query parameters clearly distinguished
+- **Type-safe**: Pydantic validation on all parameters
+- **Composable**: Query params can be mixed (endpoint-specific + Lucene filtering)
+- **Flexible**: Set parameters in any order, validation happens when accessing path
+- **Maintainable**: Base paths in `BasePath` class
+- **Less code**: ~40 lines vs ~100+ lines (including query param class)
+
+---
+
+## Centralized Path Management
+
+### Old Approach
+
+Base paths scattered across inheritance hierarchy:
+
+```python
+# api.py
+class Api:
+    def __init__(self):
+        self.api = "/appcenter/cisco/ndfc/api"
+
+# v1.py
+class V1(Api):
+    def __init__(self):
+        super().__init__()
+        self.v1 = f"{self.api}/v1"
+
+# lan_fabric.py
+class LanFabric(V1):
+    def __init__(self):
+        super().__init__()
+        self.lan_fabric = f"{self.v1}/lan-fabric"
+
+# rest.py
+class Rest(LanFabric):
+    def __init__(self):
+        super().__init__()
+        self.rest = f"{self.lan_fabric}/rest"
+
+# control.py
+class Control(Rest):
+    def __init__(self):
+        super().__init__()
+        self.control = f"{self.rest}/control"
+
+# fabrics.py
+class Fabrics(Control):
+    def __init__(self):
+        super().__init__()
+        self.fabrics = f"{self.control}/fabrics"
+```
+
+To change `/appcenter/cisco/ndfc/api` → you must edit `api.py` (but affects all endpoints).
+
+---
+
+### New Approach
+
+All base paths in ONE location:
+
+```python
+# base_paths.py
+class BasePath:
+    NDFC_API: Final = "/appcenter/cisco/ndfc/api"
```

Comment:
I like this approach and it addresses one of my initial concerns (distributed and deeply nested tree) when we started heading down this path